components/script/dom/performance.rs

```diff
@@ -273,10 +287,70 @@ impl Performance {
             o.notify();
         }
     }
-
     fn now(&self) -> f64 {
         (time::precise_time_ns() - self.navigation_start_precise).to_ms()
     }
+    fn can_add_resource_timing_entry(&self) -> bool {
+        self.resource_timing_buffer_current_size.get() <=
+            self.resource_timing_buffer_size_limit.get()
+    }
+    fn copy_secondary_resource_timing_buffer(&self) {
+        while self.can_add_resource_timing_entry() && !self
+            .resource_timing_secondary_entries
+            .borrow_mut()
+            .is_empty()
+        {
+            self.queue_entry(
+                &(*(self
+                    .resource_timing_secondary_entries
+                    .borrow_mut()
+                    .pop_front()
+                    .unwrap())),
+                true,
+            );
+        }
+    }
+    fn fire_buffer_full_event(&self) {
+        while !self
+            .resource_timing_secondary_entries
+            .borrow_mut()
+            .is_empty()
+        {
+            let no_of_excess_entries_before =
+                self.resource_timing_secondary_entries.borrow_mut().len();
+            let mut no_of_excess_entries_after =
+                self.resource_timing_secondary_entries.borrow_mut().len();
+            if !self.can_add_resource_timing_entry() {
+                self.eventtarget
+                    .fire_event(atom!("resourcetimingbufferfull"));
```

Comment:
```rust
self.upcast::<EventTarget>()
      .fire_event(atom!("resourcetimingbufferfull"));
```

Reply:
https://doc.servo.org/script/dom/index.html#inheritance and https://doc.servo.org/script/dom/bindings/inheritance/trait.Castable.html should make it more clear.